src/main/java/bio/terra/app/model/CloudRegion.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.app.model;
+
+
+import bio.terra.common.EnumUtils;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+import java.io.IOException;
+
+/**
+ * Represents regions across different cloud providers
+ */
+@JsonDeserialize(using = CloudRegion.CloudRegionDeserializer.class)
+public interface CloudRegion {
+    String name();
+
+    class CloudRegionDeserializer extends StdDeserializer<CloudRegion> {
+
+        public CloudRegionDeserializer() {
+            this(CloudRegion.class);
+        }
+        public CloudRegionDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public CloudRegion deserialize(
+            JsonParser jsonParser,
+            DeserializationContext ctxt) throws IOException, JsonProcessingException {
+            JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+            if (node.isNull()) {
+                return null;
+            }
+            if (!node.isTextual()) {
+                throw new InvalidFormatException(
+                    jsonParser,
+                    "Invalid representation of CloudRegion",
+                    node,
+                    CloudRegion.class);
+            }
+            String value = node.textValue();
+            CloudRegion region = EnumUtils.valueOfLenient(GoogleRegion.class, value);
```

Comment:
This system depends on no name collisions in the future. It would be nice to not have to add a namespace-type prefix to every enum element. This problem is often solved by having the type written as a separate field in the data, is that not possible here, e.g. if we're deserializing region and resource, do we always have a platform as well?

Reply:
We could.  FWIW, this is many many years away from being a problem (likely never).  Azure, GCP and AWS all have different naming schemes for their regions...so maybe one day we'll support Oracle cloud and they collide...but that seems...distant